source/containers/Form/Form.tsx

```diff
@@ -109,15 +107,18 @@ const Form: React.FC<Props> = ({
 
   const [hasSigned, setHasSigned] = useState(status.type.includes('signed'));
 
-  const [hasUploaded, setHasUploaded] = useState(attachments?.length && attachments.filter(({ uploadedFileName }) => uploadedFileName).length == attachments.length);
+  const [hasUploaded, setHasUploaded] = useState(
+    attachments?.length &&
```

Comment:
Outside of this PR but as the Embold comment says, maybe change to `attachments?.length > 0` while we're at it 🙂.

Reply:
I'm unsure if we should, as `.length` only checks if it's an array, and not if it has any entries.

Reply:
I see now that it might be valid for length to be 0, but in that case that check is still wrong since if length == 0 it would still evaluate to `false`.

There shouldn't be any difference from `if (arr.length)` and `if(arr.length > 0)` as any number value not 0 evaluates to `true`. I think the problem comes from doing `<number> && <bool>` as it's unclear what is intended and it might cause unwanted side-effects.

If it's valid for length to be 0 and we need to check if it's an array we should do something like `Array.isArray(...)` or `hasOwnProperty`, however that would need testing and is outside this PR so for now we can just leave it as is.

Reply:
We'll leave it as is for now.